src/lib/ProductExporter.ts

```diff
@@ -186,6 +197,27 @@ export class ProductExporter {
             }
         }
     }
+    private retrieveReferenceChargesIds(chargesIds: Array<String>){
+        let chargeReferenceIds = new Array<any>();
+        chargesIds.filter(chargeId => chargeId['enxCPQ__Charge_Reference__r'])
+                  .forEach(chargeId => {chargeReferenceIds.push(chargeId['enxCPQ__Charge_Reference__r'])});
```

Comment:
I would consider using map() instead of initializing empty array and then pushing to it